src/content/en/2020/resource-hints.md

```diff
@@ -5,19 +5,367 @@ chapter_number: 21
+  sql_file="preconnect_and_dnsprefetch_relative.sql"
+) }}
+
+Close to 2% of pages (~40k) reported the issue of both `preconnect` & `dns-prefetch` in a single resource.
+
+{{ figure_markup(
+  caption="Pages that pass the preconnect Lighthouse audit",
+  content="19.67%",
+  classes="big-number",
+  sheets_gid="152449420",
+  sql_file="lighthouse_preconnect.sql"
+) }}
+
+We saw only 19.67% of pages passing Lighthouse’s "[Preconnect to required origins](https://web.dev/uses-rel-preconnect/)" audit, creating a large opportunity for thousands of websites to start using `preconnect` or `dns-prefetch` to establish early connections to important third-party origins.
+
+{{ figure_markup(
+  caption="Pages that pass the preload Lighthouse audit",
+  content="84.6%",
+  classes="big-number",
+  sheets_gid="1047875076",
+  sql_file="lighthouse_preload.sql"
+) }}
+
+Running Lighthouse’s "[Preload key requests](https://web.dev/uses-rel-preload/)" audit resulted in 84.6% of pages passing the test, which is an astonishing result. If you are looking to use `preload` for the first time, remember, fonts and critical scripts are a good place to start.
+
+### Native Lazy Loading
+
+Now let’s celebrate the first year of the [Native Lazy Loading](https://addyosmani.com/blog/lazy-loading/) API, which at the time of publishing already has over [72%](https://caniuse.com/loading-lazy-attr) browser support. This new API can be used to defer the load of below-the-fold iframes and images on the page until the user scrolls near them. This can reduce data usage, memory usage, and helps speed up above-the-fold content. Opting-in to lazy load is as simple as adding `loading=lazy`  on `<iframe>` or `<img>` elements.
+
+{{ figure_markup(
+  caption="Percentage of pages using native lazy loading set to “lazy”",
+  content="3.87%",
+  classes="big-number",
+  sheets_gid="2039808014",
+  sql_file="native_lazy_loading_attrs.sql"
+) }}
+
+Adoption is still in its early days, especially with the official thresholds earlier this year being too conservative, and only [recently](https://addyosmani.com/blog/better-image-lazy-loading-in-chrome/) aligning with developer expectations. With almost 72% of browsers supporting native image/source lazy loading, this is another area of opportunity especially for pages looking to improve data usage and performance on low-end devices. 
+
+{{ figure_markup(
+  caption="Percentage of pages passing the offscreen images Lighthouse audit",
+  content="68.65%",
+  classes="big-number",
+  sheets_gid="1357389632",
+  sql_file="lighthouse_offscreen_images.sql"
+) }}
+
+Running Lighthouse's "[Defer offscreen images](https://web.dev/offscreen-images/)" audit resulted in 68.65% of pages passing the test. For those pages there is an opportunity to lazy-load images after all critical resources have finished loading.
+
+Be mindful to run the audit on both desktop and mobile as images may move off screen when the viewport changes.
+
+## Machine Learning
```

Comment:
What do you think about changing this title to predictive prefetching? As Machine Learning is just one way or predicting this.

```suggestion
## Predictive prefetching
```